src/test/fuzz/mini_miner.cpp

```diff
@@ -184,6 +185,11 @@ FUZZ_TARGET(mini_miner_selection, .init = initialize_miner)
     node::BlockAssembler::Options miner_options;
     miner_options.blockMinFeeRate = target_feerate;
     miner_options.nBlockMaxWeight = MAX_BLOCK_WEIGHT;
+    // Only setting reserved weight when necessary based on the template size
+    const auto reserved_weight = MAX_BLOCK_WEIGHT - pool.GetTotalTxSize();
+    if (reserved_weight < DEFAULT_BLOCK_RESERVED_WEIGHT) {
+        miner_options.block_reserved_weight = reserved_weight;
+    }
```

Comment:
I don't think this conditional check is necessary. It will be simpler and more straightforward to just set the minimum rather than creating a new variable and then checking the conditional statement.  

In the worst case, we are creating two variables here, whereas in my suggestion, we are just updating the initial variable.  

```suggestion  
    miner_options.block_reserved_weight = MINIMUM_BLOCK_RESERVED_WEIGHT;  
```

Reply:
Yes, this would be simpler but I am not sure if we want it to be simpler. With fuzzing we usually try to hit as different many scenarios as possible and that's why I chose this, sometimes setting `block_reserved_weight` and sometimes not, and setting it to a variety of values. I'm not an expert on fuzzing though, maybe someone with a focus on it can weight in here.

Reply:
@marcofleon @dergoegge Do you have feedback on what your preferred approach would be here?

Reply:
It seems like the conditional never ends up being true, which you can see in the test's [coverage](https://marcofleon.github.io/coverage/miniminer/coverage/root/bitcoin/src/test/fuzz/mini_miner.cpp.html). 

In general, I do think you're right about wanting the fuzz test to cover a range of possible values, but I'm not quite familar enough with the mining code to know what the best approach here would be.

Reply:
Ok, that hit an error further down below. It seems to me like the cluster limit of 500 tx was hit. Let me try again with a max of 500 tx and increasing the potential number of outputs instead.